packages/browser/src/integrations/featureFlags/featureFlagsIntegration.ts

```diff
@@ -0,0 +1,47 @@
+import type { Client, Event, EventHint, Integration, IntegrationFn } from '@sentry/core';
+
+import { defineIntegration } from '@sentry/core';
+import { copyFlagsFromScopeToEvent, insertFlagToScope } from '../../utils/featureFlags';
+
+export interface FeatureFlagsIntegration extends Integration {
+  setFlag: (name: string, value: unknown) => void;
+}
+
+/**
+ * Sentry integration for buffering feature flags manually with an API, and
+ * capturing them on error events. We recommend you do this on each flag
+ * evaluation. Flags are buffered per Sentry scope and limited to 100 per event.
+ *
+ * See the [feature flag documentation](https://develop.sentry.dev/sdk/expected-features/#feature-flags) for more information.
+ *
+ * @example
+ * ```
+ * import * as Sentry from '@sentry/browser';
+ * import { type FeatureFlagsIntegration } from '@sentry/browser';
+ *
+ * // Setup
+ * Sentry.init(..., integrations: [Sentry.featureFlagsIntegration()])
+ *
+ * // Verify
+ * const flagsIntegration = Sentry.getClient()?.getIntegrationByName<FeatureFlagsIntegration>('FeatureFlags');
+ * if (flagsIntegration) {
+ *   flagsIntegration.setFlag('my-flag', true);
+ * } else {
+ *   // check your setup
+ * }
+ * Sentry.captureException(Exception('broke')); // 'my-flag' should be captured to this Sentry event.
+ * ```
+ */
+export const featureFlagsIntegration = defineIntegration(() => {
+  return {
+    name: 'FeatureFlags',
+
+    processEvent(event: Event, _hint: EventHint, _client: Client): Event {
+      return copyFlagsFromScopeToEvent(event);
+    },
+
+    setFlag(name: string, value: unknown): void {
+      insertFlagToScope(name, value);
+    },
```

Comment:
Just conventionally, this is a very odd API and way of interacting with the package - I recommend not exposing `setFlag`, or any additional method on an integration.

Is there a way we could instead create an API like `Sentry.experimental_featureFlags.setFlagOnScope(name, value)` or something like that?

Reply:
@Lms24 just informed me that this was already discussed offline and is done to avoid bundle-size increases.

I mentioned that it is in theory possible to expose top-level APIs in the integration CDN bundles without increasing bundle size for non-FF users by basically doing `window.Sentry.setFlag = () => {...}`. This was apparently not considered so I'll suggest it here. There may be incompatibilities with other requirements though like for example `lazyLoadIntegration()` which I haven't thought through.

I also decided I'll leave this up to you and unblock. I personally find the API a bit whacky though.

Reply:
Thanks for the recommendation @lforst ! 
> window.Sentry.setFlag = () => {...}

Where would this code be added - some sort of loader script? Our decision in https://github.com/getsentry/sentry-javascript/pull/14207 was to **exclude FFs from CDN bundles entirely.**

Another concern we had about a top-level API is users who don't install this integration could still use `setFlag`. I guess we could enforce this with a check on `getClient().getIntegration()`. But for now I'd prefer to keep this as-is, so we can release quickly and test it in Sentry first (there's no docs planned for this atm)

Reply:
@lforst what's unusual about it? We do similar things w/ replay/feedback where we expose public methods on the integration. We wanted it to be opt-in, especially during alpha/beta phase. We can later decide to promote it as a core API as needed.